interp/modintern.ml

```diff
@@ -59,33 +59,42 @@ let lookup_module lqid = fst (lookup_module_or_modtype Module lqid)
 
 let transl_with_decl env = function
   | CWith_Module ((_,fqid),qid) ->
-      WithMod (fqid,lookup_module qid)
+      WithMod (fqid,lookup_module qid), Univ.ContextSet.empty
   | CWith_Definition ((_,fqid),c) ->
     let c, ectx = interp_constr env (Evd.from_env env) c in
     let ctx = UState.context ectx in
-      WithDef (fqid,(c,ctx))
+    if Flags.is_universe_polymorphism () then
+      let inst, ctx = Univ.abstract_universes ctx in
+      let c = Vars.subst_univs_level_constr (Univ.make_instance_subst inst) c in
+      WithDef (fqid,(c, Some ctx)), Univ.ContextSet.empty
+    else
+      WithDef (fqid,(c, None)), Univ.ContextSet.of_context ctx
```

Comment:
Here we have `ContextSet.of_context (UState.context ectx)` which is better done as `UState.context_set ectx`.